evalml/data_checks/data_check_action_option.py

```diff
@@ -0,0 +1,129 @@
+"""Recommended action returned by a DataCheck."""
+
+from evalml.data_checks.data_check_action_code import DataCheckActionCode
+
+
+class DataCheckActionOption:
+    """A recommended action option returned by a DataCheck.
+
+        It contains an action code that indicates what the
+        action should be, a data check name that indicates what data check was used to generate the action, and
+        parameters and metadata which can be used to further refine the action.
+
+    Args:
+        action_code (DataCheckActionCode): Action code associated with the action option.
+        data_check_name (str): Name of the data check that produced this option.
+        parameters (dict): Parameters associated with the action option. Defaults to None.
+        metadata (dict, optional): Additional useful information associated with the action option. Defaults to None.
+    """
+
+    def __init__(self, action_code, data_check_name, parameters=None, metadata=None):
+        self.action_code = action_code
+        self.data_check_name = data_check_name
+        self.parameters = parameters
```

Comment:
Might be useful to leave the structure required for what a parameter looks like (esp since there's so much validation going on)

Reply:
Agreed, I think it'd be useful to show an example of an initialized `DataCheckActionOption` object with valid  parameters